zebra-state/src/service/memory_state/non_finalized_state.rs

```diff
@@ -93,6 +93,7 @@ impl NonFinalizedState {
     pub fn any_chain_contains(&self, hash: &block::Hash) -> bool {
         self.chain_set
             .iter()
+            .rev()
```

Comment:
this isn't strictly necessary but I thought it would be best for us to iterate in best chain order when searching for hashes.